src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -75,29 +84,227 @@ public DarCollectionService(DarCollectionDAO darCollectionDAO, DarCollectionServ
    * @param userRole The role the user is making the request as
    * @return List of DarCollectionSummary objects
    */
-  public List<Object> getSummariesForRoleName(User user, String userRole) {
-    switch (userRole) {
-      case Resource.ADMIN:
-        // TODO: Find all
+  private void updateStatusCount(Map<String, Integer> statusCount, String status) {
+    Integer count = statusCount.get(status);
+    if(Objects.isNull(count)) {
+      statusCount.put(status, 0);
+      count = 0;
+    }
+    statusCount.put(status, count + 1);
+  }
+
+  private void determineCollectionStatus(DarCollectionSummary summary, Map<String, Integer> statusCount, Integer datasetCount, Integer electionCount) {
+    //If there are no elections, status is unreviewed
+    //if there are some elections open, status is in process
+    //if all elections are closed or canceled and electionCount == datasetCount, status is complete
+    if(electionCount == 0) {
+      summary.setStatus(DarCollectionStatus.UNREVIEWED.getValue());
+    } else if(electionCount == datasetCount) {
+      Integer openCount = statusCount.get(ElectionStatus.OPEN.getValue());
+      if(Objects.isNull(openCount)) {
+        summary.setStatus(DarCollectionStatus.COMPLETE.getValue());
+      } else {
+        summary.setStatus(DarCollectionStatus.IN_PROCESS.getValue());
+      }
+    } else {
+      summary.setStatus(DarCollectionStatus.IN_PROCESS.getValue());
+    }
+  }
+
+  private void processDarCollectionSummariesForAdmin(List<DarCollectionSummary> summaries) {
+    //if at least one election is open, show cancel
+    //if at least one non-open/absent election, show open
+    summaries.forEach(s -> {
+      Map<String, Integer> statusCount = new HashMap<>();
+      Map<Integer, Election> elections = s.getElections();
+      if(elections.size() == 0) {
+        s.addAction(DarCollectionActions.OPEN.getValue());
+        s.setStatus(DarCollectionStatus.UNREVIEWED.getValue());
+      } else {
+        elections.values().forEach(e -> {
+          String status = e.getStatus();
+          updateStatusCount(statusCount, status);
+          if(status == ElectionStatus.OPEN.getValue()) {
+            s.addAction(DarCollectionActions.CANCEL.getValue());
+          } else {
+            s.addAction(DarCollectionActions.OPEN.getValue());
+          }
+        });
+        determineCollectionStatus(s, statusCount, s.getDatasetCount(), s.getElections().size());
+      }
+    });
+  }
+
+  private void processDarCollectionDraftsAsSummaries(List<DataAccessRequest> drafts, List<DarCollectionSummary> summaries) {
+    drafts.forEach(d -> {
+      try{
+        DarCollectionSummary summary = new DarCollectionSummary();
+        Date createDate = new Date(d.getData().getCreateDate());
+        String darCode = "DRAFT_DAR_" + new SimpleDateFormat("yyyy-MM-dd")
+          .format(createDate).toString();
```

Comment:
nit: `format()` already returns a string
```suggestion
          .format(createDate);
```